internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -103,9 +104,29 @@ func (r *GitJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, fmt.Errorf("error retrieving git job: %v", err)
 	}
 
-	if errors.IsNotFound(err) && gitRepo.Status.Commit != "" {
-		if err := r.createJob(ctx, &gitRepo); err != nil {
-			return ctrl.Result{}, fmt.Errorf("error creating git job: %v", err)
+	if errors.IsNotFound(err) {
+		if gitRepo.Status.Commit != "" {
```

Comment:
Is there an easier way to express this logic which uses not as many nested ifs?

Maybe 

```
if commit == "" || disablePolling { updateCommit }
if commit != "" { createJob }
```